components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth/OAuthServiceTest.java

```diff
@@ -0,0 +1,522 @@
+        }
+        oAuthConsumer.setOauthSignature(signature);
+
+        whenNew(OAuthConsumerDAO.class).withAnyArguments().thenReturn(oAuthConsumerDAO);
+        when(oAuthConsumerDAO.getOAuthConsumerSecret(anyString())).thenReturn(consumerSecret);
+
+        OAuthService oAuthService = new OAuthService();
+        oAuthService.isOAuthConsumerValid(oAuthConsumer);
+    }
+
+    @Test
+    public void testGetOauthRequestToken() throws Exception {
+
+        String consumerSecret = "consumer-secret";
+
+        Parameters requestParams = new Parameters();
+        requestParams.setOauthConsumerKey("consumer-key");
+        requestParams.setOauthNonce("oauth-nonce");
+        requestParams.setOauthTimeStamp(GREATER_THAN_LATEST_TIMESTAMP.toString());
+        requestParams.setOauthSignatureMethod("HmacSHA1");
+        requestParams.setBaseString("http://is.com:8080/playground.com");
+        requestParams.setHttpMethod("HTTP-POST");
+        requestParams.setOauthCallback("http://is.com:8080/playground.com");
+        requestParams.setScope("openid");
+
+        // Create consumer signature.
+        String signature = getConsumerSignature(requestParams, consumerSecret, null);
+
+        // Set the created signature to the oAuthConsumer.
+        requestParams.setOauthSignature(signature);
+
+        prepareForValidateTimestampAndNonce(consumerSecret, false);
+
+        OAuthService oAuthService = new OAuthService();
+        Parameters responseParams = oAuthService.getOauthRequestToken(requestParams);
+        assertEquals(responseParams.getOauthConsumerKey(), requestParams.getOauthConsumerKey(), "ConsumerKey should" +
+                " be same as the given ConsumerKey.");
+        assertTrue(StringUtils.isNotBlank(responseParams.getOauthToken()), "Should generate an non-blank OAuthToken.");
+        assertTrue(StringUtils.isNotBlank(responseParams.getOauthTokenSecret()), "Should generate an non-blank " +
+                "OAuthTokenSecret.");
+    }
+
+    @Test(dataProvider = "consumerSecretValidSignature", expectedExceptions = AuthenticationException.class)
+    public void testGetOauthRequestTokenException(String consumerSecret, boolean isValidSignature) throws Exception {
+
+        // Create Parameters.
+        Parameters requestParams = new Parameters();
+        requestParams.setOauthConsumerKey("consumer-key");
+        requestParams.setOauthNonce("oauth-nonce");
+        requestParams.setOauthTimeStamp(GREATER_THAN_LATEST_TIMESTAMP.toString());
+        requestParams.setOauthSignatureMethod("HmacSHA1");
+        requestParams.setBaseString("http://is.com:8080/playground.com");
+        requestParams.setHttpMethod("HTTP-POST");
+        String signature;
+        if (isValidSignature) {
+            signature = getConsumerSignature(requestParams, consumerSecret, null);
+        } else {
+            signature = "an-invalid-signature";
+        }
+        requestParams.setOauthSignature(signature);
+
```

Comment:
Remove unnecessary new lines.

Reply:
this is a required separation. input request parameters vs preparing and calling the method under test.